contrib_bots/lib/giphy/giphy.py

```diff
@@ -0,0 +1,116 @@
+# To use this plugin, you need to set up the Giphy API key for this bot in
+# zulip_giphy_config.py
+
+from __future__ import absolute_import
+from . import zulip_giphy_config as config
+import requests
+import logging
+
+class GiphyHandler(object):
+    '''
+    This plugin posts a GIF in response to the keywords provided by the user.
+    Images are provided by Giphy, through the public API.
+    The bot looks for messages starting with "@giphy" or @mention of the bot
+    and responds with a message with the GIF based on provided keywords.
+    It also responds to private messages.
+    '''
+    def usage(self):
+        return '''
+            This plugin allows users to post GIFs provided by Giphy.
+            Users should preface keywords with "@giphy" or the Giphy-bot @mention.
+            The bot responds also to private messages.
+            '''
+
+    def get_url_gif_giphy(self, keyword, api_key):
+        # Return a URL for a Giphy GIF based on keywords given.
+        # In case of error, e.g. failure to fetch a GIF URL, it will
+        # return a number.
+        query = {'s': keyword,
+                 'api_key': api_key}
+        try:
+            data = requests.get('http://api.giphy.com/v1/gifs/translate', params=query)
+        except requests.exceptions.ConnectionError as e:  # Usually triggered by bad connection.
+            logging.warning(e)
+            return 1
+
+        if not data.status_code == 200:
```

Comment:
Could we make these 2 if statements as one? 
`if search_status != 200 and data.status_code != 200: return 1`?